components/gallery/GalleryForm.tsx

```diff
@@ -1,43 +1,49 @@
 /* eslint-disable import/no-unresolved */
 import * as React from "react";
+import BN from "bn.js";
 import Form from "react-bootstrap/Form";
 import Col from "react-bootstrap/Col";
 import Row from "react-bootstrap/Row";
 import Button from "react-bootstrap/Button";
 import InputGroup from "react-bootstrap/InputGroup";
-import { MediaGalleryItemStreamManager } from "../../lib/stream-managers/MediaGalleryItemStreamManager";
-import { useFirebase } from "../../lib/Firebase";
+import type { MediaObject, Encoding } from "@geo-web/types";
+import { AssetId, AccountId } from "caip";
 import { CID } from "multiformats/cid";
-
+import { GalleryModalProps } from "./GalleryModal";
 import {
   galleryFileFormats,
   getFormat,
   getFormatCS,
-  getFormatType,
 } from "./GalleryFileFormat";
-import { MediaGalleryStreamManager } from "../../lib/stream-managers/MediaGalleryStreamManager";
-import { MediaObject } from "schema-org-ceramic/types/MediaObject.schema";
-import { GalleryModalProps } from "./GalleryModal";
+import { MediaGalleryItem } from "../../lib/geo-web-content/mediaGallery";
+import { useFirebase } from "../../lib/Firebase";
+import { NETWORK_ID } from "../../lib/constants";
+
+interface NewMediaGalleryItem {
```

Comment:
You should also be able to use `MediaObject` here where `content` is a `CID` and can avoid the `ipfs://` parsing below.

Reply:
The problem with using `MediaObject` here is that the fields `content` and `encodingFormat` are required and so the object should be defined in one go, but unless we don't allow user input except for uploading we get the data to fill the  object one a time.
The compromise here is to let the user input strings and type checking that what we submit [here](https://github.com/Geo-Web-Project/cadastre/blob/ede60a7a6c2e8443347c7d904b1d6dbff14d943f/components/gallery/GalleryForm.tsx#L224) is actually a `MediaObject`.

Reply:
Oh, gotcha.